Ginger/GingerCore/ALM/QtestCore.cs

```diff
@@ -0,0 +1,685 @@
+                                        if (stepsStatuses.Where(x => x.Name == "Failed").Count() > 0)
+                                            testCaseStatus = statuses.Where(z => z.Name == "Failed").FirstOrDefault();
+                                        else if (stepsStatuses.Where(x => x.Name == "No Run").Count() == testStepsCount || stepsStatuses.Where(x => x.Name == "Not Applicable").Count() == testStepsCount)
+                                            testCaseStatus = statuses.Where(z => z.Name == "Unexecuted").FirstOrDefault();
+                                        else if (stepsStatuses.Where(x => x.Name == "Passed").Count() == testStepsCount || (stepsStatuses.Where(x => x.Name == "Passed").Count() + stepsStatuses.Where(x => x.Name == "Not Applicable").Count()) == testStepsCount)
+                                            testCaseStatus = statuses.Where(z => z.Name == "Passed").FirstOrDefault();
+                                        else
+                                            testCaseStatus = statuses.Where(z => z.Name == "Unexecuted").FirstOrDefault();
+
+                                        QTestApiModel.ManualTestLogResource automationTestLog = new QTestApiModel.ManualTestLogResource(null, null, bizFlow.StartTimeStamp, bizFlow.EndTimeStamp,
+                                                                                                                                                null, null, tsTest.TestName + " - execution", null, null,
+                                                                                                                                                null, null, null, testCaseStatus, null, testStepLogs);
+
+                                        QTestApiModel.TestLogResource testLog = testlogApi.SubmitTestLog((long)Convert.ToInt32(ALMCore.DefaultAlmConfig.ALMProjectKey), automationTestLog, (long)Convert.ToInt32(tsTest.Runs[0].RunID));
+
+                                        // Attach ActivityGroup Report if needed                                       
+                                        if (publishToALMConfig.ToAttachActivitiesGroupReport)
+                                        {
+                                            if ((activGroup.TempReportFolder != null) && (activGroup.TempReportFolder != string.Empty) &&
+                                                (System.IO.Directory.Exists(activGroup.TempReportFolder)))
+                                            {
+                                                //Creating the Zip file - start
+                                                string targetZipPath = System.IO.Directory.GetParent(activGroup.TempReportFolder).ToString();
+                                                string zipFileName = targetZipPath + "\\" + TestCaseName.ToString() + "_GingerHTMLReport.zip";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [There's no need to call 'ToString()' on a string.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5395796)